diff -ur boost_1_53_0/boost/token_functions.hpp boost_1_53_0-fixed/boost/token_functions.hpp
--- boost_1_53_0/boost/token_functions.hpp	2012-12-11 18:42:26.000000000 +0400
+++ boost_1_53_0/boost/token_functions.hpp	2016-02-20 14:46:28.739422110 +0300
@@ -370,7 +370,7 @@
         return_partial_last_(return_partial_last) { }
 
     offset_separator()
-      : offsets_(1,1), current_offset_(),
+      : offsets_(1,1.), current_offset_(), //icpc-specific workaround
         wrap_offsets_(true), return_partial_last_(true) { }
 
     void reset() {
diff -ur boost_1_53_0/boost/xpressive/regex_token_iterator.hpp boost_1_53_0-fixed/boost/xpressive/regex_token_iterator.hpp
--- boost_1_53_0/boost/xpressive/regex_token_iterator.hpp	2012-09-18 21:53:58.000000000 +0400
+++ boost_1_53_0/boost/xpressive/regex_token_iterator.hpp	2016-02-20 14:47:27.874755542 +0300
@@ -95,7 +95,7 @@
 
 inline std::vector<int> to_vector(int subs)
 {
-    return std::vector<int>(1, subs);
+    return std::vector<int>(1., subs); //icpc-specific workaround
 }
 
 inline std::vector<int> const &to_vector(std::vector<int> const &subs)
