--- dyninst-20171106/cmake/shared.cmake	2017-10-12 13:55:20.690896079 -0500
+++ dyninst-20171106-fixes/cmake/shared.cmake	2017-11-07 21:20:02.359255315 -0600
@@ -99,7 +99,7 @@
 include (${DYNINST_ROOT}/cmake/optimization.cmake)
 
 # Check for cotire-gcc compatibility
-set(USE_COTIRE true)
+set(USE_COTIRE false)
 IF(CMAKE_COMPILER_IS_GNUCC)
     execute_process(COMMAND ${CMAKE_C_COMPILER} -dumpversion OUTPUT_VARIABLE GCC_VERSION)
     string(REGEX MATCHALL "[0-9]+" GCC_VERSION_COMPONENTS ${GCC_VERSION})
