--- dyninst-20171003/symtabAPI/src/dwarfWalker.C	2017-10-12 13:55:20.852896058 -0500
+++ dyninst-20171003-fixes/symtabAPI/src/dwarfWalker.C	2017-10-12 13:51:04.925928829 -0500
@@ -741,30 +741,18 @@
     auto highlow = parseHighPCLowPC(dbg, die);
     if(highlow.second) newRanges.push_back(highlow.first);
 
-    Address range_offset;
-    if (findConstant(DW_AT_ranges, range_offset, die, dbg))
+    Dwarf_Addr base;
+    Dwarf_Addr start, end;
+    ptrdiff_t offset = 0;
+    while(1)
     {
-        Dwarf_Aranges *ranges = NULL;
-        size_t ranges_length = 0;
-        dwarf_printf("calling ranges_a, offset 0x%lx, die %p\n", range_offset, die);
-        int status = dwarf_getaranges(dbg, &ranges, &ranges_length);
-        bool done = (status != 0);
-        for (unsigned i = 0; i < ranges_length && !done; i++) {
-            Dwarf_Arange *cur = dwarf_onearange(ranges,i);
-            Address cur_base = range_base;
-            Dwarf_Addr address;
-            Dwarf_Word length;
-            Dwarf_Off offset;
-            if(dwarf_getarangeinfo(cur, &address, &length, &offset)!=0)
-                continue;
-            Address low = address + offset + cur_base;
-            Address high = low + length;
-            dwarf_printf("Lexical block from 0x%lx to 0x%lx\n", low, high);
-            newRanges.push_back(AddressRange(low, high));
-
-        }
+        offset = dwarf_ranges(&die, offset, &base, &start, &end);
+        if(offset < 0) return newRanges;
+        newRanges.push_back(AddressRange(start, end));
+        if(offset == 0) break;
     }
     return newRanges;
+
 }
 
 bool DwarfWalker::parseLexicalBlock() {
--- dyninst-20171003/cmake/shared.cmake	2017-10-12 13:55:20.690896079 -0500
+++ dyninst-20171003-fixes/cmake/shared.cmake	2017-10-12 13:55:41.977893349 -0500
@@ -99,7 +99,7 @@
 include (${DYNINST_ROOT}/cmake/optimization.cmake)
 
 # Check for cotire-gcc compatibility
-set(USE_COTIRE true)
+set(USE_COTIRE false)
 IF(CMAKE_COMPILER_IS_GNUCC)
     execute_process(COMMAND ${CMAKE_C_COMPILER} -dumpversion OUTPUT_VARIABLE GCC_VERSION)
     string(REGEX MATCHALL "[0-9]+" GCC_VERSION_COMPONENTS ${GCC_VERSION})
--- dyninst-20171003/symtabAPI/src/Object-elf.C	2017-10-12 13:55:20.849896059 -0500
+++ dyninst-20171003-fixes/symtabAPI/src/Object-elf.C	2017-10-12 13:51:04.925928829 -0500
@@ -4380,7 +4380,7 @@
         auto tmp = strrchr(filename, '/');
         if(truncateLineFilenames && tmp)
         {
-            strings->push_back(tmp);
+            strings->push_back(++tmp);
         }
         else
         {
