--- mrnet-20161003/src/lightweight/Packet.c	2016-10-03 11:50:00.000000000 -0700
+++ mrnet-20161003-fixes/src/lightweight/Packet.c	2017-02-06 16:17:15.492156089 -0800
@@ -11,6 +11,7 @@
 #include "mrnet_lightweight/Network.h"
 #include "mrnet_lightweight/Packet.h"
 #include "xplat_lightweight/vector.h"
+#include "stdbool.h"
 
 void delete_Packet_t(Packet_t* packet)
 {
--- mrnet-20161003/conf/Makefile.in	2016-10-03 11:49:59.000000000 -0700
+++ mrnet-20161003-fixes/conf/Makefile.in	2017-02-06 16:18:46.192152107 -0800
@@ -145,7 +145,7 @@
 
    endif # SILENCE_WARNINGS
 
-   COMMNODE_LDFLAGS += -static-libstdc++ -static-libgcc
+   COMMNODE_LDFLAGS += 
 
 endif #GNU options
 
