--- xerces-c-3.1.1/Makefile.in	2010-04-21 15:55:00.000000000 +0000
+++ xerces-c-3.1.1-fixes/Makefile.in	2012-02-12 22:40:28.211326630 +0000
@@ -289,7 +289,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = src tests samples
+SUBDIRS = src tests 
 EXTRA_DIST = version.incl \
              config/pretty-make \
              scripts/createBindingsFormatFromHPP.pl \
